test/connectors/v2/PenaltyDetailsConnectorSpec.scala

```diff
@@ -41,29 +42,94 @@ class PenaltyDetailsConnectorSpec extends SpecBase {
         ArgumentMatchers.any(),
         ArgumentMatchers.any())
         (ArgumentMatchers.any(),
           ArgumentMatchers.any(),
           ArgumentMatchers.any()))
-        .thenReturn(Future.successful(HttpResponse(Status.OK, "")))
+        .thenReturn(Future.successful(Right(GetPenaltyDetailsSuccessResponse(mockGetPenaltyDetailsModel))))
 
-      val result: HttpResponse = await(connector.getPenaltyDetails("VATC/VRN/123456789")(HeaderCarrier()))
-      result.status shouldBe Status.OK
+      val result: GetPenaltyDetailsResponse = await(connector.getPenaltyDetails("VATC/VRN/123456789")(HeaderCarrier()))
+      result.isRight shouldBe true
     }
 
-    s"return a 400 when the call fails" in new Setup {
-      when(mockHttpClient.GET[HttpResponse](ArgumentMatchers.eq("/FOO/BAR/123456789"),
+    s"return a 404 when the call fails for Not Found" in new Setup {
+      when(mockHttpClient.GET[GetPenaltyDetailsResponse](ArgumentMatchers.eq("/FOO/BAR/123456789"),
         ArgumentMatchers.any(),
         ArgumentMatchers.any())
         (ArgumentMatchers.any(),
           ArgumentMatchers.any(),
           ArgumentMatchers.any()))
-        .thenReturn(Future.successful(HttpResponse(Status.BAD_REQUEST, "")))
+        .thenReturn(Future.successful(Left(GetPenaltyDetailsFailureResponse(Status.NOT_FOUND))))
 
-      val result: HttpResponse = await(connector.getPenaltyDetails("FOO/BAR/123456789")(HeaderCarrier()))
-      result.status shouldBe Status.BAD_REQUEST
+      val result: GetPenaltyDetailsResponse = await(connector.getPenaltyDetails("FOO/BAR/123456789")(HeaderCarrier()))
+      result.isLeft shouldBe true
+    }
+
+    s"return a 400 when the call fails for Bad Request" in new Setup {
+      when(mockHttpClient.GET[GetPenaltyDetailsResponse](ArgumentMatchers.eq("/FOO/BAR/123456789"),
+        ArgumentMatchers.any(),
+        ArgumentMatchers.any())
+        (ArgumentMatchers.any(),
+          ArgumentMatchers.any(),
+          ArgumentMatchers.any()))
+        .thenReturn(Future.successful(Left(GetPenaltyDetailsFailureResponse(Status.BAD_REQUEST))))
+
+      val result: GetPenaltyDetailsResponse = await(connector.getPenaltyDetails("FOO/BAR/123456789")(HeaderCarrier()))
+      result.isLeft shouldBe true
+    }
+
+    s"return a 409 when the call fails for Conflict" in new Setup {
+      when(mockHttpClient.GET[GetPenaltyDetailsResponse](ArgumentMatchers.eq("/FOO/BAR/123456789"),
+        ArgumentMatchers.any(),
+        ArgumentMatchers.any())
+        (ArgumentMatchers.any(),
+          ArgumentMatchers.any(),
+          ArgumentMatchers.any()))
+        .thenReturn(Future.successful(Left(GetPenaltyDetailsFailureResponse(Status.CONFLICT))))
+
+      val result: GetPenaltyDetailsResponse = await(connector.getPenaltyDetails("FOO/BAR/123456789")(HeaderCarrier()))
+      result.isLeft shouldBe true
+    }
+
+    s"return a 422 when the call fails for Unprocessed Entity" in new Setup {
```

Comment:
Could you change this slightly?
```suggestion
    s"return a 422 when the call fails for Unprocessable Entity" in new Setup {
```